pkg/destroy/powervs/powervs.go

```diff
@@ -74,15 +68,31 @@ type User struct {
 	generation int    `default:"2"`
 }
 
-func fetchUserDetails(bxSession *bxsession.Session, generation int) (*User, error) {
-	config := bxSession.Config
+func authenticateAPIKey(apikey string) (string, error) {
+	a, err := core.NewIamAuthenticatorBuilder().SetApiKey(apikey).Build()
+	if err != nil {
+		return "", err
+	}
+	token, err := a.GetToken()
+	if err != nil {
+		return "", err
+	}
+	return token, nil
+}
+
+func fetchUserDetails(apikey string) (*User, error) {
 	user := User{}
 	var bluemixToken string
 
-	if strings.HasPrefix(config.IAMAccessToken, "Bearer") {
-		bluemixToken = config.IAMAccessToken[7:len(config.IAMAccessToken)]
+	iamToken, err := authenticateAPIKey(apikey)
+	if err != nil {
+		return &user, err
+	}
+
+	if strings.HasPrefix(iamToken, "Bearer") {
+		bluemixToken = iamToken[7:]
 	} else {
-		bluemixToken = config.IAMAccessToken
+		bluemixToken = iamToken
```

Comment:
Can't you reuse this from the `session.go` file in the installconfig dir?

Reply:
@hamzy can you give input here? I think it would be good to reuse code if we can but maybe you have rationale for keeping it separate

Reply:
Sure reusing code is always good.

Reply:
Code refactored, force-pushed. PTAL